cdap-api/src/main/java/co/cask/cdap/api/schedule/RunRequirements.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.schedule;
+
+import java.util.Objects;
+import javax.annotation.Nullable;
+
+/**
+ * Defines requirements that must be satisfied at runtime in order for the scheduler to launch a run.
+ *
+ * Currently only contains the maximum number of concurrent runs. In the future, other checks may be added,
+ * such as the amount of available memory in the YARN cluster.
+ */
+public class RunRequirements {
```

Comment:
Is RunConstraints a better name? It describes limitations rather than requirements. But that is just my take...